includes/base/controls-stack.php

```diff
@@ -1081,6 +1081,34 @@ public function get_active_settings( $settings = null, $controls = null ) {
 				}
 
 				$active_settings[ $setting_key ] = $setting;
+
+				// Enabling assets loading from the registered control fields.
+				if ( ! empty( $control['assets'] ) ) {
+					foreach ( $control['assets'] as $assets_type => $dependencies ) {
+						foreach ( $dependencies as $dependency ) {
+							if ( ! empty( $dependency['conditions'] ) ) {
+								$is_condition_fulfilled = Conditions::check( $dependency['conditions'], $settings );
+
+								if ( ! $is_condition_fulfilled ) {
+									continue;
+								}
+							}
+
+							Plugin::$instance->assets_loader->enable_asset( $assets_type, $dependency['name'] );
+						}
+					}
+				}
+
+				// Enabling assets loading from the control object.
+				$control_conditional_assets = $control_obj::get_assets( $setting );
+
+				if ( $control_conditional_assets ) {
+					foreach ( $control_conditional_assets as $assets_type => $dependencies ) {
+						foreach ( $dependencies as $dependency ) {
+							Plugin::$instance->assets_loader->enable_asset( $assets_type, $dependency );
+						}
+					}
+				}
```

Comment:
Move to the saving process, and save as a meta with the dependent assets.

The meta will be saved to `_elementor_assets`

Add an upgrade script to delete the old `_elementor_css` meta.

The `Plugin::$instance->files_manager->clear_cache()` will delete both `_elementor_assets` & `_elementor_css`. until v3.6.